core/trino-main/src/main/java/io/trino/metadata/DiscoveryNodeManager.java

```diff
@@ -87,14 +90,17 @@
     @GuardedBy("this")
```

Comment:
Looks like this is now `@GuardedBy("allNodesLock")` too? Make sure that all accesses are in fact guarded by that lock and not synchronized anymore